### PR TITLE
[NPM] add npm docker images apps

### PR DESCRIPTION
**Pull request by nplanel**
What does this PR do?
---------------------

Add npm docker images apps

Which scenarios this will impact?
-------------------

Motivation
----------

Additional Notes
----------------
linked to this [PR](https://github.com/DataDog/datadog-agent/pull/22769)

### CODE REVIEW

**Review by pducolin — APPROVED** [review]

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -56,6 +56,10 @@ jobs:
             image: redis-client
           - app: tracegen
             image: tracegen
+          - app: npm-tools
```
[the remark this answers is not in the document]
Comment (nplanel) [reply .github/workflows/docker-publish.yml:59]: there are 2 different images used for npm tests
 * go-httpbin for the server side, deployed via docker compose
 * npm-tools that contain client tools like curl, ab, dig ...

[the remark this answers is not in the document]
Comment (nplanel) [reply .github/workflows/docker-publish.yml:59]: the deployment happen [here](https://github.com/DataDog/test-infra-definitions/pull/612/files#diff-5b21991be47c2922383bdc0b6bf00b65af7db51b82d049dd8d6ad03e3d37ac98R105)

[the remark this answers is not in the document]
Comment (leeavital) [reply .github/workflows/docker-publish.yml:59]: Right, but you have `app: npm-tools` specified twice -- isn't one supposed to be go-httpbin ?

**Pull request merged**